### DISCUSSION TITLE
Issue with Migrating from 3.x to 4.x - Missing Features

### DISCUSSION THREAD

**Discussion by jakubjtk** [taiga-family/taiga-ui#9971] [General]
Hi,

I'm working on migrating our application from version 3.x to 4.x, and I've encountered some challenges. In the previous version, I relied on several features that now seem to be missing or have changed. Unfortunately, I haven't been able to find any information in the documentation or changelog about how to use them now or what to replace them with.

- TuiCheckedDirective
- TuiFocusedDirective
- TuiPressedDirective

Thank you in advance for your help! 😊

**Comment by waterplea** [comment]
These directives are gone as we no longer needed them — they were kind of low level and undocumented. You would need to copy-paste them into your code base:

- [TuiCheckedDirective](https://github.com/taiga-family/taiga-ui/blob/v3.x/projects/cdk/directives/checked/checked.directive.ts)
- [TuiFocusedDirective](https://github.com/taiga-family/taiga-ui/blob/v3.x/projects/cdk/directives/focused/focused.directive.ts)
- [TuiPressedDirective](https://github.com/taiga-family/taiga-ui/blob/v3.x/projects/cdk/directives/pressed/pressed.directive.ts) this one would need a few utils copied too, some of them are gone — we completely moved to CSS `:active` in terms of pressed status styling.

**Comment by jakubjtk** [comment]
Thank you for the information and clarification! I appreciate the explanation and the suggestions. I'll review how to adapt or replace these directives in my codebase.
